File: js/src/maplestory/api/dto/character/characterItemEquipmentAddOptionDto.ts
import { CharacterItemEquipmentAddOptionBody } from '../../response/character/characterItemEquipmentBody';

/**
 * 캐릭터 장비 추가 옵션 정보
 */
class CharacterItemEquipmentAddOptionDto {
  /**
   * STR
   */
  str: string;

  /**
   * DEX
   */
  dex: string;

  /**
   * INT
   */
  int: string;

  /**
   * LUK
   */
  luk: string;

  /**
   * 최대 HP
   */
  maxHp: string;

  /**
   * 최대 MP
   */
  maxMp: string;

  /**
   * 공격력
   */
  attackPower: string;

  /**
   * 마력
   */
  magicPower: string;

  /**
   * 방어력
   */
  armor: string;

  /**
   * 이동속도
   */
  speed: string;

  /**
   * 점프력
   */
  jump: string;

  /**
   * 보스 공격 시 데미지 증가(%)
   */
  bossDamage: string;

  /**
   * 데미지(%)
   */
  damage: string;

  /**
   * 올스탯(%)
   */
  allStat: string;

  /**
   * 착용 레벨 감소
   */
  equipmentLevelDecrease: number;

  constructor(obj: CharacterItemEquipmentAddOptionBody) {
    const {
      str,
      dex,
      int,
      luk,
      max_hp,
      max_mp,
      attack_power,
      magic_power,
      armor,
      speed,
      jump,
      boss_damage,
      damage,
      all_stat,
      equipment_level_decrease,
    } = obj;

    this.str = str;
    this.dex = dex;
    this.int = int;
    this.luk = luk;
    this.maxHp = max_hp;
    this.maxMp = max_mp;
    this.attackPower = attack_power;
    this.magicPower = magic_power;
    this.armor = armor;
    this.speed = speed;
    this.jump = jump;
    this.bossDamage = boss_damage;
    this.damage = damage;
    this.allStat = all_stat;
    this.equipmentLevelDecrease = equipment_level_decrease;
  }
}

export { CharacterItemEquipmentAddOptionDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterItemEquipmentAddOptionDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * 附加道具選項
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterItemEquipmentAddOptionDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterItemEquipmentAddOptionDTO {

    /**
     * 力量 (STR)
     */
    @JsonProperty("str")
    private String strength;

    /**
     * 敏捷 (DEX)
     */
    @JsonProperty("dex")
    private String dexterity;

    /**
     * 敏捷 (DEX)
     */
    @JsonProperty("int")
    private String intelligence;

    /**
     * 幸運 (LUK)
     */
    @JsonProperty("luk")
    private String luck;

    /**
     * 最大體力 (HP)
     */
    @JsonProperty("max_hp")
    private String maxHp;

    /**
     * 最大 MP
     */
    @JsonProperty("max_mp")
    private String maxMp;

    /**
     * 攻擊力
     */
    @JsonProperty("attack_power")
    private String attackPower;

    /**
     * 魔力
     */
    @JsonProperty("magic_power")
    private String magicPower;

    /**
     * 防禦力
     */
    @JsonProperty("armor")
    private String armor;

    /**
     * 移動速度
     */
    @JsonProperty("speed")
    private String speed;

    /**
     * 跳躍力
     */
    @JsonProperty("jump")
    private String jump;

    /**
     * 對首領傷害增加 (%)
     */
    @JsonProperty("boss_damage")
    private String bossDamage;

    /**
     * 傷害 (%)
     */
    @JsonProperty("damage")
    private String damage;

    /**
     * 全部能力值 (%)
     */
    @JsonProperty("all_stat")
    private String allStat;

    /**
     * 裝備等級減少
     */
    @JsonProperty("equipment_level_decrease")
    private Long equipmentLevelDecrease;
}